bluetooth: stm32wb: any BT_STM32_IPM application fails to build for stm32wb5mmg — IPCC IRQs invisible to automatic NUM_IRQS computation

### Describe the bug

### Describe the bug

Building `samples/bluetooth/hci_uart` for the in-tree board `stm32wb5mmg` fails at
link time:

```
gen_isr_tables.py: error: IRQ 45 (offset=0) exceeds the maximum of 40
```

This is the exact command sequence that the `stm32wb5mmg` board documentation
instructs users to run in order to bring up Bluetooth on the B-U585I-IOT02A
Discovery kit, so the documented bring-up path for that board is currently
broken.

**Root cause**

Commit 71aa3be37bcd22cc06eebaf8400ce5255ffa92d2 ("soc: st: stm32: compute
CONFIG_NUM_IRQS automatically") made `CONFIG_NUM_IRQS` derive from the highest
interrupt number among devicetree-enabled nodes:

```kconfig
# soc/st/stm32/Kconfig.defconfig
DT_NVIC_PATH := /soc/interrupt-controller@e000e100
DT_NVIC_MAX_IRQN := $(dt_highest_controller_irq_number,$(DT_NVIC_PATH),irq)

configdefault NUM_IRQS
	default $(inc,$(DT_NVIC_MAX_IRQN)) if $(dt_path_enabled,$(DT_NVIC_PATH))
```

For the lean `hci_uart` image the highest devicetree-visible IRQ is 40 (EXTI),
giving `CONFIG_NUM_IRQS=41`.

However, `CONFIG_BT_STM32_IPM` is enabled for this target, and
`drivers/bluetooth/hci/ipm_stm32wb.c` connects the IPCC interrupts using
hardcoded CMSIS enums rather than devicetree:

```c
/* drivers/bluetooth/hci/ipm_stm32wb.c */
IRQ_CONNECT(IPCC_C1_RX_IRQn, 0, HW_IPCC_Rx_Handler, NULL, 0);   /* IRQ 44 */
IRQ_CONNECT(IPCC_C1_TX_IRQn, 0, HW_IPCC_Tx_Handler, NULL, 0);   /* IRQ 45 */
```

The `st,stm32wb-rf` node has no `interrupts` property, and there is no `ipcc`
node in `dts/arm/st/wb/stm32wb.dtsi`. IRQs 44 and 45 are therefore invisible to
the automatic computation, and the generated vector table is five entries too
small.

**Why this cannot be worked around by the user**

`CONFIG_NUM_IRQS` is declared promptless (`arch/arm/core/Kconfig`):

```kconfig
config NUM_IRQS
	int
```

so `-DCONFIG_NUM_IRQS=...` and `.conf` fragment assignments are ignored. The
value can only be changed from a Kconfig file.

### Regression

Introduced by 71aa3be37bcd22cc06eebaf8400ce5255ffa92d2. Verified by building
`samples/bluetooth/hci_uart` for `stm32wb5mmg` at that commit and at its parent,
with `west update` run at each revision so module revisions matched the
manifest:

| Revision | Result | `CONFIG_NUM_IRQS` |
|----------|--------|-------------------|
| `aea7120dab1f` (parent) | builds, links `zephyr.elf` | 63 |
| `71aa3be37bcd` | `gen_isr_tables.py` error, IRQ 45 > max 40 | 41 |
| `850f0cb04713` (`main`, today) | same failure | 41 |



### Regression

- [x] This is a regression.

### Steps to reproduce



```bash
west build -b stm32wb5mmg samples/bluetooth/hci_uart --pristine
```

### Relevant log output

```shell
[173/176] Generating isr_tables.c, isr_tables_vt.ld, isr_tables_swi.ld
FAILED: [code=1] zephyr/isr_tables.c zephyr/isr_tables_vt.ld zephyr/isr_tables_swi.ld
gen_isr_tables.py: error: IRQ 45 (offset=0) exceeds the maximum of 40
ninja: build stopped: subcommand failed.
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- Zephyr: `main` @ 850f0cb047132b10a665508aae4ab8e0a88e3663 (v4.4.99)
- Board: `stm32wb5mmg` (BLE module on B-U585I-IOT02A)
- Host OS: macOS (arm64)
- Toolchain: Zephyr SDK

### Additional Context

**Scope: `stm32wb5mmg` specifically, not the whole STM32WB series**

Other STM32WBX boards are unaffected, because they enable enough devicetree
nodes that the computed `NUM_IRQS` happens to land above 45 anyway:

| Board | Sample | `CONFIG_NUM_IRQS` | Result |
|-------|--------|-------------------|--------|
| `stm32wb5mmg` | `bluetooth/hci_uart` | 41 | fails |
| `stm32wb5mmg` | `bluetooth/peripheral_hr` | 41 | fails |
| `nucleo_wb55rg` | `bluetooth/hci_uart` | 54 | builds |
| `nucleo_wb55rg` | `bluetooth/peripheral_hr` | 54 | builds |
| `stm32wb5mm_dk` | `bluetooth/peripheral_hr` | 52 | builds |

So the failure is not sample-specific — any `CONFIG_BT_STM32_IPM` application
fails on `stm32wb5mmg` — but it is board-specific in practice. `stm32wb5mmg` is
a bare module target with a minimal devicetree, so the highest `status = "okay"`
interrupt is EXTI at 40, whereas the richer Nucleo/DK boards enable peripherals
with higher interrupt numbers.

This is worth emphasising because it means the other boards are only passing by
coincidence. The full STM32WB55 vector table extends to IRQ 62 (`dmamux1`), and
`nucleo_wb55rg` gets 54 — so any of these targets could regress the moment a
devicetree node is disabled. The correctness of `NUM_IRQS` on STM32WB currently
depends on unrelated peripherals happening to be enabled.

**Missing test coverage**

`samples/bluetooth/hci_uart/tests.yaml` defines 12 scenarios, and every one of
them restricts `platform_allow` to Nordic platforms
(`nrf52dk`, `nrf52840dk`, `nrf21540dk`, `nrf54l15dk`, `nrf54lm20dk`,
`nrf52833dk`, `nrf5340dk` cpuapp/cpunet, `nrf5340_audio_dk`, `nrf52_bsim`).
No STM32 platform appears in the file, and `samples/bluetooth/hci_uart/boards/`
contains no STM32 overlays. Since `platform_allow` is a whitelist, twister never
builds this sample for `stm32wb5mmg`, which is why CI does not catch this.

`boards/st/stm32wb5mmg/stm32wb5mmg.yaml` places no `testing:` restrictions on
the board, so the gap is specific to the sample's whitelist.

**Suggested fix**

The cleanest fix consistent with the direction of 71aa3be37bc would be to make
the IPCC interrupts visible to devicetree — either by adding an `ipcc` node to
`dts/arm/st/wb/stm32wb.dtsi`, or by giving the `st,stm32wb-rf` node an
`interrupts` property — and switching `ipm_stm32wb.c` to `DT_INST_IRQ()` instead
of the CMSIS `IPCC_C1_*_IRQn` enums. That removes the hardcoded IRQ numbers and
lets the existing automatic computation produce the correct result.

A narrower alternative is a `configdefault` override in
`soc/st/stm32/stm32wbx/Kconfig.defconfig` (series files are `rsource`d before
the family defaults, and `configdefault` is first-match-wins):

```kconfig
configdefault NUM_IRQS
	default 46 if BT_STM32_IPM
```

This is what I am using locally as a workaround, and it does resolve the build.
I note however that a similar per-series approach was proposed and not merged in
#107584, so the devicetree fix is probably preferable.

It would also be worth adding a `build_only` `stm32wb5mmg` scenario to
`samples/bluetooth/hci_uart/tests.yaml` so this combination cannot regress
silently again.

**Related**

- #107585 — same class of problem on STM32H7 (closed)
- #107584 — per-series `configdefault` fix for STM32H7 (closed, not merged)
- #107674 — proposal to give `CONFIG_NUM_IRQS` a prompt, by the author of
  71aa3be37bc, noting the promptless oversight (closed, not merged)
- #107808 / #108023 — migration-guide note on setting promptless Kconfig values
  (merged)

Those all concerned a user's own application needing a larger `NUM_IRQS`. This
report is different in that an in-tree board, in-tree sample and in-tree driver
do not build together, with no user configuration involved.

--- a/drivers/bluetooth/hci/ipm_stm32wb.c
+++ b/drivers/bluetooth/hci/ipm_stm32wb.c
@@ -22,6 +22,14 @@
 #include "shci.h"
 #include "shci_tl.h"
 
+#define STM32_IPCC_RX_IRQ	DT_INST_IRQ_BY_NAME(0, rx, irq)
+#define STM32_IPCC_RX_IRQ_PRIO	DT_INST_IRQ_BY_NAME(0, rx, priority)
+#define STM32_IPCC_TX_IRQ	DT_INST_IRQ_BY_NAME(0, tx, irq)
+#define STM32_IPCC_TX_IRQ_PRIO	DT_INST_IRQ_BY_NAME(0, tx, priority)
+
+BUILD_ASSERT(STM32_IPCC_RX_IRQ == IPCC_C1_RX_IRQn, "Unexpected IRQ number for IPCC Rx");
+BUILD_ASSERT(STM32_IPCC_TX_IRQ == IPCC_C1_TX_IRQn, "Unexpected IRQ number for IPCC Tx");
+
 static const struct stm32_pclken clk_cfg[] = STM32_DT_CLOCKS(DT_DRV_INST(0));
 
 #define POOL_SIZE (CFG_TLBLE_EVT_QUEUE_LENGTH * 4 * \
@@ -396,8 +404,8 @@ void ipcc_reset(void)
 		LL_IPCC_CHANNEL_4 | LL_IPCC_CHANNEL_5 | LL_IPCC_CHANNEL_6);
 
 	/* Set IPCC default IRQ handlers */
-	IRQ_CONNECT(IPCC_C1_RX_IRQn, 0, HW_IPCC_Rx_Handler, NULL, 0);
-	IRQ_CONNECT(IPCC_C1_TX_IRQn, 0, HW_IPCC_Tx_Handler, NULL, 0);
+	IRQ_CONNECT(STM32_IPCC_RX_IRQ, STM32_IPCC_RX_IRQ_PRIO, HW_IPCC_Rx_Handler, NULL, 0);
+	IRQ_CONNECT(STM32_IPCC_TX_IRQ, STM32_IPCC_TX_IRQ_PRIO, HW_IPCC_Tx_Handler, NULL, 0);
 }
 
 void transport_init(void)

--- a/dts/arm/st/wb/stm32wb.dtsi
+++ b/dts/arm/st/wb/stm32wb.dtsi
@@ -666,6 +666,9 @@
 		compatible = "st,stm32wb-rf";
 		clocks = <&rcc STM32_CLOCK(AHB3, 20)>,
 			 <&rcc STM32_SRC_LSE RFWKP_SEL(1)>;
+		interrupt-parent = <&nvic>;
+		interrupts = <44 0>, <45 0>;
+		interrupt-names = "rx", "tx";
 	};
 
 	smbus1: smbus1 {

--- a/dts/bindings/bluetooth/st,stm32wb-rf.yaml
+++ b/dts/bindings/bluetooth/st,stm32wb-rf.yaml
@@ -11,9 +11,23 @@ include: bt-hci.yaml
 properties:
   clocks:
     required: true
+
   bt-hci-name:
     default: "BT IPM"
+
   bt-hci-bus:
     default: "ipc"
+
   bt-hci-quirks:
     default: ["no-reset"]
+
+  interrupts:
+    required: true
+    # Do not set min-len/max-len due to DT parsing process issues.
+    # Rely on interrupt-names for the consistency.
+
+  interrupt-names:
+    required: true
+    enum: ["rx", "tx"]
+    min-len: 2
+    max-len: 2

